calico_containers/adapter/datastore.py

```diff
@@ -716,6 +792,51 @@ def get_ep_id_from_cont(self, hostname, container_id):
                 "Container with ID %s has no endpoints." % container_id)
 
     @handle_errors
+    def query_endpoint(self, endpoint_id=None, workload_id=None, orchestrator_id=None, hostname=None):
```

Comment:
This function is essentially a more functional `get_endpoints` which doesn't require that all parameters are provided. It behaves fine when used by the new `endpoint show` command, other parts of calicoctl which rely on the current `get_endpoints` command break when they are mmoved over to this new one. I need to investigate further how this is different from `get_endpoints`